lib/atlas/graph_from_yaml.rb

```diff
@@ -33,20 +29,35 @@ def build_graph
 
     private
 
-    def set_slots(node, node_attributes)
-      node_attributes.each_pair do |direction, slot_attributes|
-        slot_collection = node.public_send("#{ direction }_slots")
+    def graph_nodes
+      Hash[@graph.nodes.map do |node|
+        attributes       = @graph_yaml.fetch(:nodes)[node.key] || {}
+        node_attributes  = attributes.slice!(:in, :out)
```

Comment:
I believe this destroys part of the original @graph_yaml hash. Better `slice` and `except` separately - also more readable :wink: